src/containers/viewSingleRequest/index.tsx

```diff
@@ -0,0 +1,185 @@
+  const requestId = Number(
+    useParams<{ request_id: string; user_id: string }>().request_id,
+  );
+  const userId = Number(
+    useParams<{ request_id: string; user_id: string }>().user_id,
+  );
+  const [contacts, setContacts] = useState<AsyncRequest<ContactInfo, any>>(
+    AsyncRequestNotStarted(),
+  );
+  const [decision, setDecision] = useState<AsyncRequest<void, any>>(
+    AsyncRequestNotStarted(),
+  );
+  const [status, setStatus] = useState<Status>(Status.PENDING);
+
+  const denyRequest = () => {
+    setDecision(AsyncRequestLoading());
+    protectedApiClient
+      .denyPFRequest(requestId)
+      .then((res) => {
+        setDecision(AsyncRequestCompleted(res));
+        setStatus(Status.DENIED);
+      })
+      .catch((error) => {
+        setDecision(AsyncRequestFailed(error));
+      });
+  };
+
+  const approveRequest = () => {
+    setDecision(AsyncRequestLoading());
+    protectedApiClient
+      .approvePFRequest(requestId)
+      .then((res) => {
+        setDecision(AsyncRequestCompleted(res));
+        setStatus(Status.APPROVED);
+      })
+      .catch((error) => {
+        setDecision(AsyncRequestFailed(error));
+      });
+  };
+
+  useEffect(() => {
+    if (asyncRequestIsNotStarted(contacts)) {
+      setContacts(AsyncRequestLoading());
+      protectedApiClient
+        .getContactInfoById(userId)
+        .then((c) => {
+          setContacts(AsyncRequestCompleted(c));
+        })
+        .catch((error) => {
+          setContacts(AsyncRequestFailed(error));
+        });
+    }
+  }, [contacts, userId]);
+
+  if (status === Status.APPROVED && asyncRequestIsComplete(contacts)) {
+    return (
+      <DecisionConfirmation
+        approved={true}
+        contactInformation={contacts.result}
+      />
+    );
+  } else if (status === Status.DENIED && asyncRequestIsComplete(contacts)) {
+    return (
+      <DecisionConfirmation
+        approved={false}
+        contactInformation={contacts.result}
+      />
+    );
+  } else {
+    return (
+      <>
+        <Helmet>
+          <title>View Family Details</title>
+          <meta
+            name="description"
+            content="Specific details about a registered family."
+          />
+        </Helmet>
+        <ChungusContentContainer>
+          <LinkButton type="link" to={'/view-requests/'}>
+            &lt; Back to Requests
+          </LinkButton>
+          {asyncRequestIsComplete(contacts) && (
+            <ContactInfoSummary userId={userId} info={contacts.result} />
+          )}
+          {asyncRequestIsLoading(contacts) && <Spin />}
+          {asyncRequestIsFailed(contacts) && (
+            <StyledAlert
+              message="Error"
+              description={contacts.error.message}
+              type="error"
+              showIcon
+            />
+          )}
+        </ChungusContentContainer>
+        {asyncRequestIsLoading(decision) ? (
+          <Spin />
+        ) : (
+          <DecisionButtons>
```

Comment:
💡 Maybe what we could do here is disable these buttons when the most recent decision is not pending? And display some text like: 'this family does not have a pending request to review' or something?